browser/resources/brave_new_tab_page_refresh/models/backgrounds.ts

```diff
@@ -3,62 +3,35 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-export type BackgroundType = 'brave' | 'custom' | 'solid' | 'gradient'
-
-export interface BraveBackground {
-  type: 'brave'
-  author: string
-  imageUrl: string
-  link: string
-}
-
-export interface ColorBackground {
-  type: 'solid' | 'gradient'
-  cssValue: string
+import { NewTabPageAdEventType } from 'gen/brave/components/brave_ads/core/mojom/brave_ads.mojom.m.js'
+
+import {
+  BraveBackground,
+  SponsoredImageBackground,
+  SelectedBackground,
+  SelectedBackgroundType } from 'gen/brave/browser/ui/webui/brave_new_tab_page_refresh/brave_new_tab_page.mojom.m.js'
+
+export {
+  BraveBackground,
+  NewTabPageAdEventType,
+  SponsoredImageBackground,
+  SelectedBackgroundType
 }
 
-export interface CustomBackground {
-  type: 'custom'
-  imageUrl: string
-}
-
-export interface SponsoredImageLogo {
-  alt: string
-  destinationUrl: string
-  imageUrl: string
-}
-
-export interface SponsoredImageBackground {
-  type: 'sponsored-image' | 'sponsored-rich-media'
-  imageUrl: string
-  creativeInstanceId: string
-  campaignId: string
-  wallpaperId: string
-  shouldMetricsFallbackToP3a: boolean
-  logo: SponsoredImageLogo | null
-}
-
-export type SponsoredRichMediaEventType =
-  'click' |
-  'interaction' |
-  'mediaPlay' |
-  'media25' |
-  'media100'
-
 export type Background =
-  BraveBackground |
-  ColorBackground |
-  CustomBackground |
-  SponsoredImageBackground
+  { type: 'brave' } & BraveBackground |
```

Comment:
We should explore whether this union type can be moved to mojo as part of an effort to improve the background-related page handler interface.